crates/polars-stream/src/nodes/in_memory_sink.rs

```diff
@@ -51,8 +57,18 @@ impl ComputeNode for InMemorySink {
     }
 
     fn finalize(&mut self) -> PolarsResult<Option<DataFrame>> {
-        let mut morsels_per_pipe = core::mem::take(&mut *self.morsels_per_pipe.get_mut());
+        let morsels_per_pipe = core::mem::take(&mut *self.morsels_per_pipe.get_mut());
         let dataframes = linearize(morsels_per_pipe);
-        Ok(Some(accumulate_dataframes_vertical_unchecked(dataframes)))
+        if dataframes.is_empty() {
+            let cols = self
```

Comment:
There is a `impl From<&Schema> for DataFrame`

Reply:
Not anymore :) It's now `DataFrame::empty_with_schema`, but am using it now.